provider/pkg/provider/provider.go

```diff
@@ -1583,10 +1577,41 @@ func (k *kubeProvider) Diff(ctx context.Context, req *pulumirpc.DiffRequest) (*p
 				newInputs.GetNamespace(), newInputs.GetName())
 		}
 
-		for k, v := range detailedDiff {
-			switch v.Kind {
-			case pulumirpc.PropertyDiff_ADD_REPLACE, pulumirpc.PropertyDiff_DELETE_REPLACE, pulumirpc.PropertyDiff_UPDATE_REPLACE:
-				replaces = append(replaces, k)
+		// Remove any ignored changes from the computed diff.
+		var ignorePaths []resource.PropertyPath
+		for _, ignore := range req.IgnoreChanges {
+			ignorePath, err := resource.ParsePropertyPath(ignore)
+			if err != nil {
+				return nil, fmt.Errorf("failed to parse ignore path: %w", err)
+			}
+			ignorePaths = append(ignorePaths, ignorePath)
+		}
+		if len(ignorePaths) > 0 {
+			var diffPaths []resource.PropertyPath
+			for p := range detailedDiff {
+				diffPath, err := resource.ParsePropertyPath(p)
+				if err != nil {
+					return nil, fmt.Errorf("failed to parse diff path: %w", err)
+				}
+				diffPaths = append(diffPaths, diffPath)
+			}
+			for _, ignorePath := range ignorePaths {
+				for _, diffPath := range diffPaths {
+					if ignorePath.Contains(diffPath) {
+						delete(detailedDiff, diffPath.String())
```

Comment:
Note I have found some instances where resource.PropertyPath doesn't retain the format when calling `.String()`. I will have fixes for this as part of https://github.com/pulumi/pulumi/pull/10095.